openmc/mgxs_library.py

```diff
@@ -1634,6 +1681,285 @@ def set_inverse_velocity_mgxs(self, inverse_velocity, temperature=294.,
+                        new_shape = (num_polar, num_azimuthal) + \
+                            orig_data.shape
+                        new_data = np.resize(orig_data, new_shape)
+
+                    setter = getattr(xsdata, 'set_' + xs)
+                    setter(new_data, temp)
+
+        return xsdata
+
+    def convert_scatter_format(self, target_format, target_order=None):
+        """Produce a new MGXSLibrary object with the same data, but converted
+        to the new scatter format and order
+
+        Parameters
+        ----------
+        target_format : {'tabular', 'legendre', 'histogram'}
+            Representation of the scattering angle distribution
+        target_order : int
+            Either the Legendre target_order, number of bins, or number of
+            points used to describe the angular distribution associated with
+            each group-to-group transfer probability
+
+        Returns
+        -------
+        openmc.XSdata
+            Multi-group cross section data with the same data as in self, but
+            represented as specified in :param:`target_format`.
+
+        """
+
+        from scipy.interpolate import interp1d
+        from scipy.integrate import simps
+        from scipy.special import eval_legendre
+
+        check_value('target_format', target_format, _SCATTER_TYPES)
+        check_type('target_order', target_order, Integral)
+        if target_format == 'legendre':
+            check_greater_than('target_order', target_order, 0, equality=True)
+        else:
+            check_greater_than('target_order', target_order, 0)
+
+        xsdata = copy.deepcopy(self)
+        xsdata.scatter_format = target_format
+        xsdata.order = target_order
+
+        # Reset and re-generate XSdata.xs_shapes with the new scattering format
+        xsdata._xs_shapes = None
+
+        for i, temp in enumerate(xsdata.temperatures):
+            orig_data = self._scatter_matrix[i]
+            new_shape = orig_data.shape[:-1] + (xsdata.num_orders,)
+            new_data = np.zeros(new_shape)
+
+            if self.scatter_format == 'legendre':
+                if target_format == 'legendre':
+                    # Then we are changing orders and only need to change
+                    # dimensionality of the mu data and pad/truncate as needed
+                    order = min(xsdata.num_orders, self.num_orders)
+                    new_data[..., :order] = orig_data[..., :order]
+
+                elif target_format == 'tabular':
+                    mu = np.linspace(-1, 1, xsdata.num_orders)
+                    # Evaluate the legendre on the mu grid
+                    for imu in range(len(mu)):
+                        new_data[..., imu] = \
+                            np.sum((l + 0.5) * eval_legendre(l, mu[imu]) *
+                                   orig_data[..., l]
+                                   for l in range(self.num_orders))
+                elif target_format == 'histogram':
```

Comment:
Add a blank line between `if-else` blocks here to correspond to the convention below (and to make more readable).